src/AnalyticsEventReporter.ts

```diff
@@ -1,38 +1,78 @@
 export class AnalyticsEventReporter implements AnalyticsEventService {
-
     private config: AnalyticsConfig;
     private payload?: EventPayload;
-
     /**
      * @param config - necessary analytics config: Must provide one and only
      * one of API Key or Bearer Token.
-     * 
+     *
      * @param payload - (optional) desired event values to report
      */
     constructor(config: AnalyticsConfig, payload?: EventPayload) {
-        const apiKeyIsSet = config.key !== undefined;
-        const bearerTokenIsSet = config.bearer !== undefined;
-        const configIsValid = (apiKeyIsSet || bearerTokenIsSet) && !(apiKeyIsSet && bearerTokenIsSet);
-        if (!configIsValid) {
-            throw new Error("Provide one and only one of API Key or Bearer Token.")
-        }
-        this.config = config;
-        this.payload = payload;
+      const apiKeyIsSet = config.key !== undefined;
+      const bearerTokenIsSet = config.bearer !== undefined;
+      const configIsValid = (apiKeyIsSet || bearerTokenIsSet) && !(apiKeyIsSet && bearerTokenIsSet);
+      if (!configIsValid) {
+        throw new Error('Provide one and only one of API Key or Bearer Token.');
+      }
+      this.config = config;
+      this.payload = payload;
     }
 
     with(payload: EventPayload): AnalyticsEventService {
-        const currentPayload = this.payload === undefined ? payload : merge(this.payload, payload);
-        return new AnalyticsEventReporter(this.config, currentPayload);
+      const currentPayload = this.payload === undefined ? payload : merge(this.payload, payload);
+      return new AnalyticsEventReporter(this.config, currentPayload);
     }
 
-    report(payload: object): Promise<EventAPIResponse> | boolean {
-        return false;
+    public async report(newPayload?: PartialPayload): Promise<boolean | EventAPIResponse> {
+      const finalPayload = this.payload && newPayload
+        ? merge(this.payload, newPayload)
+        : this.payload ?? newPayload ?? function(){throw Error('EventPayload is empty');}();
+
+      /** If session tracking is enabled, and sessionId was not already manually added to the event payload,
+       * call getOrSetupSessionId to set value.
+       */
+      if (this.config.sessionTrackingEnabled && !finalPayload.sessionId) {
+        finalPayload.sessionId = getOrSetupSessionId();
+      }
+
+      finalPayload.clientSdk
+        ? (finalPayload.clientSdk as Record<string, string>)[name] = version
+        : finalPayload.clientSdk = {
+          [name]: version,
+        };
+
+      finalPayload.authorization = this.config.key
+        ? 'KEY ' + this.config.key
+        : 'Bearer ' + this.config.bearer;
+
+      const res = await post(
+        setupRequestUrl(
+          this.config.env,
+          this.config.region),
+        finalPayload as EventPayload,
+        this.config.forceFetch);
+
+
+      if (typeof res === 'boolean') {
+        return res;
+      } else if (!res.ok) {
```

Comment:
I believe we know its safe as we call await on the post call. Then if ok is undefined or does not have a property, it would be falsy, and the error case would execute.